### PR TITLE
docs: Document ACL org unit mode behavior [TECH-1590]

### PR DESCRIPTION
**Pull request by muilpp**

### CODE REVIEW

File: releases/2.41/README.md [hunk 1]
```diff
@@ -40,6 +40,23 @@ The `index` field has been removed from the `objectsReport` as the objects are n
 The `orgUnitName` field has been removed from `GET /tracker/enrollments` and `GET /tracker/events` endpoints, so it is not anymore possible to order on this field.
 The `followup` field has been renamed to `followUp` in the response for `GET /tracker/events` CSV endpoint.
 
+#### ACL tracker export breaking changes
```
[the remark this answers is not in the document]
Comment (enricocolasante) [reply releases/2.41/README.md:43]: When describing a breaking change, it is important to make very clear what was the previous behavior and what is the current behavior.
In all the following points, it is clear what is the current behavior but without the description of the previous one, it is not so clear how it is a breaking change.

Additionally, would be good to reference the Jira ticket for each of the points if it makes sense.

**Review by enricocolasante — CHANGES_REQUESTED** [review]

File: releases/2.41/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (muilpp) [reply releases/2.41/README.md:43]: it now reflects whether each change affects older versions or not

File: releases/2.41/README.md [hunk 2]
```diff
@@ -40,6 +40,28 @@ The `index` field has been removed from the `objectsReport` as the objects are n
...
+
+5. Tracker Exporter Endpoint Responses
+    * [TECH-1630](https://dhis2.atlassian.net/browse/TECH-1630): A request to `/events` and `/tracker/events` with the organization unit mode `CHILDREN` will now produce a response comprising elements from the requested organization unit and its immediate children. Previously, it did not include events from the supplied organization unit, only its children were present in the response. This change is effective from version 2.38 onwards.
+
+    * [TECH-1656](https://dhis2.atlassian.net/browse/TECH-1656) A request to `/trackedEntities` will now result in a `ForbiddenException` if the user lacks access to the requested program or tracked entity type. In the past, this scenario would trigger an `IllegalQueryException`.
```
[the remark this answers is not in the document]
Comment (enricocolasante) [reply releases/2.41/README.md:63]: Yes, but maybe it is better to specify the HTTP code that the client will receive in the response.
The client won't see the type of exception.
I would also change in the other places `BadRequestException` with 400 response status code or something similar

**Review by enricocolasante — COMMENTED** [review]
Just a comment on the status code vs exception naming.
Lets see what @Bekkalizer has to say on the sensitive point there

File: releases/2.41/README.md [hunk 3]
```diff
@@ -40,6 +40,28 @@ The `index` field has been removed from the `objectsReport` as the objects are n
...
+
+3. Default Organisation Unit Modes
+    * [TECH-1588](https://dhis2.atlassian.net/browse/TECH-1588): When neither the organization unit nor the organization unit mode is specified in the request, the default mode will be `ACCESSIBLE`. In contrast, older versions of `/trackedEntities` and `/enrollments` would return an exception when neither was specified. The SELECTED mode will continue being the default when an organization unit is specified.
+
+4. Organization Unit Mode ALL Authorization
+    * [TECH-1589](https://dhis2.atlassian.net/browse/TECH-1589): In `/enrollments` and `/tracker/enrollments`, the organization unit mode `ALL` is now restricted to users with either `ALL` or `F_TRACKED_ENTITY_INSTANCE_SEARCH_IN_ALL_ORGUNITS` authorities, consistent with the other two endpoints. Previously, any user could use the `ALL` mode, even if it might not return any results based on the user scope. This change is effective from version 2.38 onwards.
```
[the remark this answers is not in the document]
Comment (Bekkalizer) [reply releases/2.41/README.md:56]: Not sensitive as such - the user does not get access to more data. It does change the behavior for the API slightly, as we are now giving an error on calls that would not previously.

**Review by Bekkalizer — APPROVED** [review]

**Review by enricocolasante — APPROVED** [review]

**Pull request merged**